pywbem/cim_http.py

```diff
@@ -445,551 +233,161 @@ def wbem_request(url, data, creds, cimxml_headers=None, debug=False, x509=None,
-                    ctx.load_cert_chain(self.cert_file, keyfile=self.key_file)
-                if self.ca_certs:
-                    # We need to use CERT_REQUIRED to require that the server
-                    # certificate is being validated by the client (against the
-                    # certificates in ca_certs).
-                    ctx.verify_mode = SSL.CERT_REQUIRED
-                    if os.path.isdir(self.ca_certs):
-                        ctx.load_verify_locations(capath=self.ca_certs)
-                    else:
-                        ctx.load_verify_locations(cafile=self.ca_certs)
-                    ctx.check_hostname = True
-                else:
-                    ctx.check_hostname = False
-                    ctx.verify_mode = SSL.CERT_NONE
-
-                # setup the socket
-                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-                sock.settimeout(self.timeout)
-
-                try:
-                    self.sock = ctx.wrap_socket(sock,
-                                                server_hostname=self.host)
-                    return self.sock.connect((self.host, self.port))
-
-                except SSLError as arg:
-                    raise ConnectionError(
-                        _format("SSL error {0}: {1}; OpenSSL version: {2}",
-                                arg.__class__, arg, OPENSSL_VERSION),
-                        conn_id=conn_id)
-                except CertificateError as arg:
-                    raise ConnectionError(
-                        _format("SSL certificate error {0}: {1}; "
-                                "OpenSSL version: {2}",
-                                arg.__class__, arg, OPENSSL_VERSION),
-                        conn_id=conn_id)
-
-    class FileHTTPConnection(HTTPBaseConnection, httplib.HTTPConnection):
-        """Execute client connection based on a unix domain socket. """
-
-        def __init__(self, uds_path):
-            httplib.HTTPConnection.__init__(self, host='localhost')
-            self.uds_path = uds_path
-
-        def connect(self):
-            try:
-                socket_af = socket.AF_UNIX
-            except AttributeError:
-                raise ConnectionError(
-                    _format("file URLs not supported on {0} platform due to "
-                            "missing AF_UNIX support", platform.system()),
-                    conn_id=conn_id)
-            self.sock = socket.socket(socket_af, socket.SOCK_STREAM)
-            self.sock.connect(self.uds_path)
-
-    if not cimxml_headers:
+    if cimxml_headers is None:
         cimxml_headers = []
 
-    host, port, use_ssl = parse_url(_ensure_unicode(url))
-
-    key_file = None
-    cert_file = None
-
-    if use_ssl and x509 is not None:
-        cert_file = x509.get('cert_file')
-        key_file = x509.get('key_file')
-
-    local_auth_header = None
+    target = '/cimom'
+    target_url = '{}{}'.format(conn.url, target)
```

Comment:
Here is where I made temp change to use:  See separate comment since this was just a workaround to continue tests.


    target = '/cimom'
    host, port, ssl = parse_url(conn.url)
    target_url = '{}:{}{}'.format(host, port, target)

Reply:
Has been done in WBEMConnection ctor.